### ISSUE TITLE
`equalalways` is smarter on splitting than closing

### ISSUE THREAD

**Issue by ubitux** [vim/vim#5955]
`vim --clean -c "vs | vs | sp | res +15 | winc l | winc l | sp | clo"`

```
What I got:       What I expected:

┌───┬───┬───┐     ┌───┬───┬───┐
│   │   │   │     │   │   │   │
├───┤   │   │     │   │   │   │
│   │   │   │     ├───┤   │   │
└───┴───┴───┘     └───┴───┴───┘
```

`equalalways` is on by default, and if you don't do the last close (`vim --clean -c "vs | vs | sp | res +15 | winc l | winc l | sp"`) you realize the last split on the right doesn't trigger a global window equalization affecting the left split:

```
┌───┬───┬───┐
│   │   │   │
│   │   ├───┤
├───┤   │   │
└───┴───┴───┘
```

 This is a good thing: the left side should not be affected by the creation of a split on the opposite side. On the other hand, closing does affect it for some reason.

Reproducible with Vim `8.2.0510`

**Comment by tyru** [comment]
Maybe `vim --clean -c "vs | vs | sp | res +15 | setl winfixheight | winc l | winc l | sp | clo"` saves you?

**Comment by ubitux** [comment]
> Maybe `vim --clean -c "vs | vs | sp | res +15 | setl winfixheight | winc l | winc l | sp | clo"` saves you?

Yeah no worry, I have workarounds; I'm reporting because I believe there is an issue. BTW, the problem with the workaround your propose is that it breaks the automatic equalization when it matters in the future (typically if I `:sp` again on the left side).
